test_pickle.py

```diff
@@ -7,4 +13,25 @@ def test_pickle(self):
             pickle.dump(data, f)
         with open('data.pickle', 'rb') as f:
             data = pickle.load(f)
-        assert data == {'a': 1, 'b': 2, 'c': 3}
\ No newline at end of file
+        assert data == {'a': 1, 'b': 2, 'c': 3}
+
+    def test_pickle_from_other_version(self, protocol=pickle.DEFAULT_PROTOCOL,os_type=False,version_number=False):
+
+        #If os is not given, we will use the running os and version
+        running_os_type,running_version_number = get_os_and_version()
+        if not os_type:
+            os_type = running_os_type
+        if not version_number:
+            version_number = running_version_number
+
+        # Checking how many test cases we have for the protocol
+        count = 0
+        for file in os.listdir(f"logs/{os_type}/{version_number}/protocol_{protocol}"):
+            count += 1
+        count = count // 2
```

Comment:
I don't get what count is doing here. I understand we increment, but why the floor division and how is that useful for the following lines?  It might be because we "duplicate" files, but selecting only some files with regex seems like the more correct decision to make.